File: 프로그래머스/unrated/181919. 콜라츠 수열 만들기/콜라츠 수열 만들기.py
def solution(n):
    answer = [n]
    while True:
        if answer[-1] == 1:
            return answer
        if answer[-1]%2 == 0:
            answer.append(answer[-1]//2)
        else:
            answer.append(answer[-1]*3+1)
    

File: 프로그래머스/0/181919. 콜라츠 수열 만들기/콜라츠 수열 만들기.java
import java.util.*;

class Solution {
    public int[] solution(int n) {
        List<Integer> result = new ArrayList<>();
        result.add(n);
        while (n != 1) {
            if(n%2 == 0) {
                n /= 2;
            } else {
                n = 3*n + 1;
            }
            result.add(n);
        }        
        return result.stream().mapToInt(v -> v).toArray();
    }
}